trace-forward/src/Trace/Forward/Network/Forwarder.hs

```diff
@@ -0,0 +1,138 @@
+connectToAcceptor iomgr config@ForwarderConfiguration{..} loQueue =
+  case acceptorEndpoint of
+    LocalPipe localPipe -> do
+      let snocket = localSnocket iomgr localPipe
+          address = localAddressFromPath localPipe
+      doConnectToAcceptor snocket address noTimeLimitsHandshake app
+    RemoteSocket host port -> do
+      acceptorAddr:_ <- Socket.getAddrInfo Nothing (Just $ T.unpack host) (Just $ show port)
+      let snocket = socketSnocket iomgr
+          address = Socket.addrAddress acceptorAddr
+      doConnectToAcceptor snocket address timeLimitsHandshake app
+ where
+  app =
+    OuroborosApplication $ \_connectionId _shouldStopSTM ->
+      [ MiniProtocol
+          { miniProtocolNum    = MiniProtocolNum 1
+          , miniProtocolLimits = MiniProtocolLimits { maximumIngressQueue = maxBound }
+          , miniProtocolRun    = forwardTraceObjects config loQueue
+          }
+      ]
+
+doConnectToAcceptor
+  :: Snocket IO fd addr
+  -> addr
+  -> ProtocolTimeLimits (Handshake UnversionedProtocol Term)
+  -> OuroborosApplication 'InitiatorMode addr LBS.ByteString IO () Void
+  -> IO ()
+doConnectToAcceptor snocket address timeLimits app =
+  connectToNode
+    snocket
+    unversionedHandshakeCodec
+    timeLimits
+    (cborTermVersionDataCodec unversionedProtocolDataCodec)
+    nullNetworkConnectTracers
+    acceptableVersion
+    (simpleSingletonVersions
+       UnversionedProtocol
+       UnversionedProtocolData
+       app)
+    Nothing
+    address
+
+forwardTraceObjects
+  :: (CBOR.Serialise lo,
+      ShowProxy lo)
+  => ForwarderConfiguration lo
+  -> TBQueue lo
+  -> RunMiniProtocol 'InitiatorMode LBS.ByteString IO () Void
+forwardTraceObjects config loQueue =
+  InitiatorProtocolOnly $
+    MuxPeerRaw $ \channel -> do
+      cv <- newEmptyTMVarIO
+      (r, trailing) <-
+        runPeerWithLimits
+          (forwarderTracer config)
+          (Forwarder.codecTraceForward CBOR.encode CBOR.decode
+                                       CBOR.encode CBOR.decode
+                                       CBOR.encode CBOR.decode)
+          (byteLimitsTraceForward (fromIntegral . LBS.length))
+          timeLimitsTraceForward
+          channel
+          (Forwarder.traceForwarderPeer $ readItems config loQueue)
+      atomically $ putTMVar cv r
+      return ((), trailing)
+
+forwardTraceObjectsResp
+  :: (CBOR.Serialise lo,
+      ShowProxy lo)
+  => ForwarderConfiguration lo
+  -> TBQueue lo
+  -> RunMiniProtocol 'ResponderMode LBS.ByteString IO Void ()
+forwardTraceObjectsResp config loQueue =
+  ResponderProtocolOnly $
+    MuxPeerRaw $ \channel -> do
+      cv <- newEmptyTMVarIO
+      (r, trailing) <-
+        runPeerWithLimits
+          (forwarderTracer config)
+          (Forwarder.codecTraceForward CBOR.encode CBOR.decode
+                                       CBOR.encode CBOR.decode
+                                       CBOR.encode CBOR.decode)
+          (byteLimitsTraceForward (fromIntegral . LBS.length))
```

Comment:
What is the type of `fromIntegral` that is used here? `LBS.length` returns `Int64` if I am not mistaken.  I am worried about an overflow error.

Reply:
I took it from `KeepAlive` protocol.